### PR TITLE
DOC: tweak colormap docs in pyplot.colormaps

### LINKED ISSUE #8923: Slightly imprecise doc wording
http://matplotlib.org/devdocs/api/pyplot_summary.html#matplotlib.pyplot.colormaps says
> The base colormaps are derived from those of the same name provided with Matlab:

and goes on to include the new colormaps viridis, plasma, magma and inferno in the list, which are not Matlab-derived, of course.

### PR DESCRIPTION
**Pull request by tacaswell**
Re-arrange the docstring to:

 - put the perceptually uniform ones at the top and promote them
 - move color brewer above the MATLAB ones
 - de-document the `jet()` and `hot()` helper methods. 

closes #8923

### CODE REVIEW

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -2041,6 +1991,57 @@ def colormaps():
...
+                  for sepia tone colorization of photographs
+      prism       repetitive red-yellow-green-blue-purple-...-green pattern
+                  (not cyclic at endpoints)
+      spring      linearly-increasing shades of magenta-yellow
+      summer      sequential linearly-increasing shades of green-yellow
+      winter      linearly-increasing shades of blue-green
+      =========   =======================================================
+
+    A set of palettes from the `Yorick scientific visualisation
+    package <http://dhmunro.github.io/yorick-doc/>`_, an evolution of
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: Can make this https now.

**Comment by tacaswell** [comment]
Third times the charm?!

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -1919,66 +1919,17 @@ def colormaps():
...
-                    blackbody radiation from an iron bar as it grows hotter
-      gist_ncar     pseudo-spectral black-blue-green-yellow-red-purple-white
-                    colormap from National Center for Atmospheric
-                    Research [#]_
-      gist_rainbow  runs through the colors in spectral order from red to
-                    violet at full saturation (like *hsv* but not cyclic)
-      gist_stern    "Stern special" color table from Interactive Data
-                    Language software
-      ============  =======================================================
+      ========    ===================================================
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: One more `=` on the first column here (and overwrite the space.)

**Comment by anntzer** [comment]
thanks!

**Pull request merged**